### ISSUE TITLE
Control and Status Register issues

### ISSUE THREAD

**Issue by TheThirdOne** [TheThirdOne/rars#2]
There are a few issues with the control and status register implementation. All of these are solvable, but there is a little awkwardness involved with each.

  - CSRs are only accessible by number in the instruction set. This can be fixed with gratuitous psuedo-ops or defining a new type of register token that can be a number or a name.
  - CSRs can not have read only sections. Many CSRs such as uip have bits which cannot be modified (but change value in hardware). The adding read only stuff should be pretty easy, but also making it so it updates properly is where some awkwardness comes in. 
  - RDCYCLE[H], RDTIME[H], RDINSTRET[H] don't exist. RDCYCLE AND RDINSTRET should be easy to implement given the above issue is done. RDTIME is a bit more problematic, because it is not clear if it should always be updating or not. It shouldn't be a big issue though; these should not be commonly used.

**Comment by Sustrak** [comment]
Realted with CSR instructions I have found a bug with the compilation of the instruction CSRW. Given the following code:
```
csrrw t1, 0x0, t0
csrrw t3, 0x0, t2
```

it should be compiled to (compiled with riscv32-unknown-elf-as):

```
00029373          	csrrw	t1,ustatus,t0
00039e73          	csrrw	t3,ustatus,t2
```

but with the RARS compiler we get:

```
00501373          	csrrw	t1,ustatus,t0
00701e73          	csrrw	t3,ustatus,t2
```

**Comment by TheThirdOne** [comment]
Thanks for the report. I think I have fixed the problem in c762e55295261f3dfbd7f8f337f2e496cad26116.

All of the CSRR instructions had the first and second arguments flipped.

**Comment by TheThirdOne** [comment]
I don't know of any remaining issues with the way CSRs are handled. The Zicsr extension should now be fully and correctly implemented.

The issue of RDTIME still exists, but that is not an issue with the CSR part, but rather with the time counter.
